driver-core/src/main/java/com/datastax/driver/core/querybuilder/BuiltStatement.java

```diff
@@ -296,18 +296,19 @@ public Boolean isIdempotent() {
     @Override
     public String toString() {
         try {
+            if (forceNoValues)
+                return getQueryString();
+            // 1) try first with all values inlined (will not work if some values require custom codecs,
+            // or if the required codecs are registered in a different CodecRegistry instance than the default one)
+            return maybeAddSemicolon(buildQueryString(null, CodecRegistry.DEFAULT_INSTANCE)).toString();
+        } catch (RuntimeException e1) {
```

Comment:
Catching `RuntimeException` prevents other unchecked exceptions to propagate, and in particular, exceptions thrown from codecs (NPE, `InvalidTypeException`, `UnsupportedOperationException`...)